isisd/isis_spf.c

```diff
@@ -1270,11 +1269,6 @@ static int isis_spf_preload_tent_ip_reach_cb(const struct prefix *prefix,
 	prefix_copy(&ip_info.dest, prefix);
 	apply_mask(&ip_info.dest);
 
-	if (prefix->family == AF_INET)
-		vtype = VTYPE_IPREACH_INTERNAL;
-	else
-		vtype = VTYPE_IP6REACH_INTERNAL;
-
```

Comment:
It will simpler and better to keep the call to `isis_spf_add_ocal()` factorisation by adding a test after AF_INET to distinguish old and new metric. ISIS metric is old or new not none.

Reply:
Considering that ISIS metric may be in transition, there will be two invocations of isis_spf_add_local(). It can be implemented as follows: 
```
	if (prefix->family == AF_INET) {
        	if (spftree->area->newmetric)
            		vtype = VTYPE_IPREACH_TE;

        	if (spftree->area->oldmetric)
            		vtype = VTYPE_IPREACH_INTERNAL;

        	if (spftree->area->oldmetric && spftree->area->newmetric)
            		transition = true;
    	}
```